src/GafferScene/IECoreScenePreview/PrimitiveAlgo.cpp

```diff
@@ -1261,16 +1281,24 @@ void PrimitiveAlgo::transformPrimitive(
 
 		if( vecVar )
 		{
-			for( Imath::V3f &i : vecVar->writable() )
-			{
-				Canceller::check( canceller );
-				transformPrimVarValue( i, i, matrix, normalMatrix, interp );
-			}
+			std::vector< Imath::V3f >& writable = vecVar->writable();
+
+			tbb::parallel_for(
```

Comment:
parallel_for gave me about an 8X speedup in transformPrimitive - it's extremely parallelizable. I wonder if it would make sense to keep the fast implementation of transformPrimitive and put anything necessary to disable threading in FreezeTransform ... actually, wait, do we need to do anything special in FreezeTransform. If it's not using TaskCollab, it won't benefit from threading, but will it cause any problems?

Reply:
It _would_ benefit from threading even without the right cache policy, but I don't have the headspace to reason about the deadlock-inducing possibilities of it not being protected by task isolation and all the other TaskCollaboration shenanigans. This PR is about `mergePrimitives()`, so please let's keep it about that. I don't expect the FreezeTransform changes to be huge, but I would benefit from considering them separately (and it keeps you moving on the main goal - the MeshMerge node).